pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PqlQueryResource.java

```diff
@@ -0,0 +1,194 @@
+      }
+
+      // Send query to a random broker.
+      String instanceId = instanceIds.get(RANDOM.nextInt(instanceIds.size()));
+      InstanceConfig instanceConfig = _pinotHelixResourceManager.getHelixInstanceConfig(instanceId);
+      String url = "http://" + instanceConfig.getHostName().split("_")[1] + ":" + instanceConfig.getPort() + "/query";
+      return sendPQLRaw(url, pqlQuery, traceEnabled);
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while processing get request", e);
+      return QueryException.getException(QueryException.INTERNAL_ERROR, e).toString();
+    }
+  }
+
+  public String sendPostRaw(String urlStr, String requestStr, Map<String, String> headers) {
+    HttpURLConnection conn = null;
+    try {
+      /*if (LOG.isInfoEnabled()){
+        LOGGER.info("Sending a post request to the server - " + urlStr);
+      }
+
+      if (LOG.isDebugEnabled()){
+        LOGGER.debug("The request is - " + requestStr);
+      }*/
+
+      LOGGER.info("url string passed is : " + urlStr);
+      final URL url = new URL(urlStr);
+      conn = (HttpURLConnection) url.openConnection();
+      conn.setDoOutput(true);
+      conn.setRequestMethod("POST");
+      // conn.setRequestProperty("Content-Type", "application/json; charset=utf-8");
+
+      conn.setRequestProperty("Accept-Encoding", "gzip");
+
+      final String string = requestStr;
+      final byte[] requestBytes = string.getBytes("UTF-8");
+      conn.setRequestProperty("Content-Length", String.valueOf(requestBytes.length));
+      conn.setRequestProperty("http.keepAlive", String.valueOf(true));
+      conn.setRequestProperty("default", String.valueOf(true));
+
+      if (headers != null && headers.size() > 0) {
+        final Set<Entry<String, String>> entries = headers.entrySet();
+        for (final Entry<String, String> entry : entries) {
+          conn.setRequestProperty(entry.getKey(), entry.getValue());
+        }
+      }
+
+      //GZIPOutputStream zippedOutputStream = new GZIPOutputStream(conn.getOutputStream());
+      final OutputStream os = new BufferedOutputStream(conn.getOutputStream());
+      os.write(requestBytes);
+      os.flush();
+      os.close();
+      final int responseCode = conn.getResponseCode();
+
+      /*if (LOG.isInfoEnabled()){
+        LOGGER.info("The http response code is " + responseCode);
+      }*/
+      if (responseCode != HttpURLConnection.HTTP_OK) {
+        throw new IOException("Failed : HTTP error code : " + responseCode);
+      }
+      final byte[] bytes = drain(new BufferedInputStream(conn.getInputStream()));
+
+      final String output = new String(bytes, "UTF-8");
+      /*if (LOG.isDebugEnabled()){
+        LOGGER.debug("The response from the server is - " + output);
+      }*/
+      return output;
+    } catch (final Exception ex) {
+      LOGGER.error("Caught exception while sending pql request", ex);
+      Utils.rethrowException(ex);
+      throw new AssertionError("Should not reach this");
```

Comment:
I want to minimize changes from the original code. If you compare with the existing PqlQueryResource.java, you will find that the logic here is the exact same thing. Once we move to the new API, we can refine things the change it in some compatible way over time.